### PR TITLE
Query transformation for external database fixed.

### LINKED ISSUE #6381: left/right/substring from mysql table reports error
**Describe the bug**
Could not use `left` or `right` or `substring` when selecting data from mysql directly

**How to reproduce**
* Version: 19.11.2
* SQL that could run with error
```SQL
-- Exception msg for below sql : 
-- Code: 1000. DB::Exception: Received from clickhouse-server:9000. DB::Exception: mysqlxx::BadQuery: Operand should contain 1 column(s)
select 
            `id`
from mysql('host:port', 'database', 'tablename', 'username', 'password')
where 1=1 and left(stat_datetime, 10)='2019-08-07'  and advertiser_id = '12345'
```
* SQL that runs OK
```SQL
select 
	`id`
from mysql('host:port', 'database', 'tablename', 'username', 'password')
where 1=1 and advertiser_id = '12345'
```

**Expected behavior**
Both SQL should run OK

### PR DESCRIPTION
**Pull request by dimarub2000**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Bug Fix

Short description (up to few sentences):
Query transformation for MySQL, ODBC, JDBC table functions now works properly for SELECT WHERE queries with multiple AND subqueries. Fixes  #6381

### CODE REVIEW

**Comment by dimarub2000** [comment]
Tests will be added later.

File: dbms/src/Storages/transformQueryForExternalDatabase.cpp [hunk 1]
```diff
@@ -141,16 +141,12 @@ String transformQueryForExternalDatabase(
             if (function->name == "and")
             {
                 bool compatible_found = false;
-                auto new_function_and = std::make_shared<ASTFunction>();
-                auto new_function_and_arguments = std::make_shared<ASTExpressionList>();
-                new_function_and->arguments = new_function_and_arguments;
-                new_function_and->children.push_back(new_function_and_arguments);
```
Comment (alexey-milovidov) [inline dbms/src/Storages/transformQueryForExternalDatabase.cpp]: This may be important (adding functions arguments as a child).

**Comment by alexey-milovidov** [comment]
BTW, you can use unit test (`gtest_transform_query_for_external_database.cpp`).

File: dbms/src/Storages/transformQueryForExternalDatabase.cpp (hunk 1 above)
Comment (dimarub2000) [reply dbms/src/Storages/transformQueryForExternalDatabase.cpp]: Probably you are right.

Comment (dimarub2000) [reply dbms/src/Storages/transformQueryForExternalDatabase.cpp]: Actually `makeASTFunction()` does the same.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**